src/pages/api/dog/[id].js

```diff
@@ -0,0 +1,18 @@
+import { getDogById } from "../../../../server/db/actions/Dog";
+
+export default async function handler(req, res) {
+  if (req.method == "GET") {
+    try {
+      const { id } = req.query;
+      const dogs = await getDogById(id);
```

Comment:
Check if `dogs` is empty, which means there's no dog in the db that has this id. I know in the ticket I said to return an empty array, but rethinking about it this should return
```js
{
    success: false,
    message: "Unable to retrieve dog because it does not exist"
}
```
and have a status code of `404 - Not found`